File: MetodosOrdenamiento/hash/hash.js
function hashSort(arr) {
    if (arr.length === 0) return arr;
    
    const minVal = Math.min(...arr);
    const maxVal = Math.max(...arr);
    
    // Crear tabla hash
    const hashTable = {};
    
    // Insertar elementos en la tabla hash
    for (const num of arr) {
        if (hashTable[num]) {
            hashTable[num]++;
        } else {
            hashTable[num] = 1;
        }
    }
    
    // Reconstruir array ordenado
    const sortedArr = [];
    for (let num = minVal; num <= maxVal; num++) {
        if (hashTable[num]) {
            for (let i = 0; i < hashTable[num]; i++) {
                sortedArr.push(num);
            }
        }
    }
    
    return sortedArr;
}

// Generar arreglo de 10 elementos aleatorios
const arr = Array.from({length: 10}, () => Math.floor(Math.random() * 100) + 1);
console.log("Array original:", arr);

// Ordenar con Hash Sort
const sortedArr = hashSort(arr);
console.log("Array ordenado:", sortedArr);

File: MetodosOrdenamiento/hash/Hash.java
import java.util.*;

public class Hash{
    public static List<Integer> hashSort(List<Integer> arr) {
        if (arr.isEmpty()) return arr;
        
        int minVal = Collections.min(arr);
        int maxVal = Collections.max(arr);
        
        // Crear tabla hash
        Map<Integer, Integer> hashTable = new HashMap<>();
        
        // Insertar elementos en la tabla hash
        for (int num : arr) {
            hashTable.put(num, hashTable.getOrDefault(num, 0) + 1);
        }
        
        // Reconstruir lista ordenada
        List<Integer> sortedArr = new ArrayList<>();
        for (int num = minVal; num <= maxVal; num++) {
            if (hashTable.containsKey(num)) {
                for (int i = 0; i < hashTable.get(num); i++) {
                    sortedArr.add(num);
                }
            }
        }
        
        return sortedArr;
    }
    
    public static void main(String[] args) {
        // Generar arreglo de 10 elementos aleatorios
        Random random = new Random();
        List<Integer> arr = new ArrayList<>();
        
        for (int i = 0; i < 10; i++) {
            arr.add(random.nextInt(100) + 1);
        }
        
        System.out.println("Array original: " + arr);
        
        // Ordenar con Hash Sort
        List<Integer> sortedArr = hashSort(arr);
        
        System.out.println("Array ordenado: " + sortedArr);
    }
}